client/components/CompetitorNavbar.tsx

```diff
@@ -22,9 +24,20 @@ import {
     MenubarSeparator,
     MenubarTrigger,
 } from '@/components/ui/menubar';
-import { Tabs, TabsList, TabsTrigger } from '@/components/ui/tabs';
+import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs';
+import { useRouter } from 'next/dist/client/components/navigation';
+import { useState } from 'react';
 
 export default function CompetitorNavbar() {
+    const router = useRouter();
+    const [tabValue, setTabValue] = useState('text-Editor');
+
+    const handleTabChange = (value: string) => {
+        setTabValue(value);
+        if (value === 'Leaderboard') {
+            router.push('/leaderboard');
+        }
+    };
```

Comment:
We do not want to navigate to the leaderboard page here, it should just be placed where the editor was:
![screenshot](https://github.com/user-attachments/assets/b01621be-48cf-4e14-901d-f6e53d31a24e)